docs/authoring.md

```diff
@@ -0,0 +1,108 @@
+Make sure the `<TargetFramework>` for the project is `net5.0-windows10.0.19041.0`. 
+And add the following to your library's project file:
+```
+  <PropertyGroup>
+    <!-- The Windows SDK metadata will need to update as new versions release -->
+    <CsWinRTWindowsMetadata>10.0.19041.0</CsWinRTWindowsMetadata>
+    <CsWinRTComponent>true</CsWinRTComponent>
+    <CsWinRTEnableLogging>true</CsWinRTEnableLogging>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <GeneratedFilesDir Condition="'$(GeneratedFilesDir)'==''">$([MSBuild]::NormalizeDirectory('$(MSBuildProjectDirectory)', '$(IntermediateOutputPath)', 'Generated Files'))</GeneratedFilesDir>
+  </PropertyGroup>
+```
+And don't forget to include a `PackageReference` to `Microsoft.Windows.CsWinRT`. You can add the nuget feed found in the C#/WinRT README as a nuget source, and install the package in Visual Studio via the `Manage NuGet Packages for Solution` option. 
+
+
+## Known Authoring Issues
+1. There are some programs you could write as/in your component that aren't available in the Windows Runtime. 
+We are working on implementing diagnostics in our tool that will catch these errors before a winmd is generated for your component.
+
+2. Not all C# types have been mapped to runtime types, but we are working on completing this coverage. 
+
+## Using the authored component in C++
+As part of the native (C++) support, the WinRT Hosting dlls (WinRT.Host and WinRT.Host.Shim) need to be in the same folder as the native executable. 
+For now, users need a special target of their own so MSBuild can place the hosting dlls in the correct place. But soon we will implement this so that a package reference to C#/WinRT in the authored component is all that is needed.   
+
+Modifications needed: 
+  1. update the consuming app's project file (e.g. `YourApp.vcxproj`)
+  2. add a manifest file and runtimeconfig file 
+  3. add a target for copying the required DLLs.
+
+
+### Project File
+The following item group demonstrates the needed modifications to the project file (1. above):
+
+```
+<ItemGroup>
+    <!-- the winmd -->
+    <Reference Include="PosnLibrary">
+      <HintPath>..\PosnLibrary\Generated Files\PosnLibrary.winmd</HintPath>
+      <IsWinMDFile>true</IsWinMDFile>
+    </Reference>
+    
+    <!-- the runtimeconfig.json -->
+    <None Include="WinRT.Host.runtimeconfig.json">
+      <DeploymentContent Condition="'$(Configuration)|$(Platform)'=='Release|x64'">true</DeploymentContent>
+      <DeploymentContent Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">true</DeploymentContent>
+    </None>
+    
+    <!-- the manifest -->
+    <Manifest Include="PosnConsole.exe.manifest">
+      <DeploymentContent>true</DeploymentContent>
+    </Manifest>
+</ItemGroup> 
+```
+
+### Manifest and RuntimeConfig
+You'll need to author some files to assist the hosting process by the native app: `YourNativeApp.exe.manifest` and `WinRT.Host.runtimeconfig.json`. 

```

Comment:
The manifest is only required for unpackaged apps, correct? If so, start with the WinRT.Host.runtimeconfig.json. Then add a comment like: "if your app is not packaged with MSIX then be sure to include your activatable class registrations in a fusion manifest file 'YourNativeApp.exe.manifest'".

Reply:
Well even for packaged scenarios, I think we may still need to do tooling work to make it do the right thing like specify our host.

Reply:
@adambraden [how is this](https://github.com/microsoft/CsWinRT/commit/4c9c94b3c1432ad694fcf296403a5459f1e05b67)